index.html

```diff
@@ -0,0 +1,283 @@
+							<input class="submit h-norm content-center strong" id="submit" type="submit" value="Add link" disabled>
+						</div>
+					</div>
+					<div class="row h-big item-center">
+						<input class="link h-norm blue pad-r pad-l" id="link" type="text" autocomplete="off" spellcheck="false" autocorrect="off" required placeholder="https://medium.com/swlh/sexism-in-tech-an-inconvenient-truth-26df0329e39">
+					</div>
+					</form>
+				</div>
+			</section>
+			<section id="listing" class="listing">
+				<table>
+					<thead>
+						<tr>
+							<th class="flex-7-5 strong silver">State</th>
+							<th class="flex-80 left strong silver">Link</th>
+							<th class="flex-12-5 strong silver">Added</th>
+						</tr>
+					</thead>
+					<tbody>
+					</tbody>
+				</table>
+
+			</section>
+		</main>
+
+	</body>
+	<script type="text/javascript">
+		// URL validation function, links that pass and don't pass listed here https://stackoverflow.com/a/15855457
+		// FAILS ON 'https://www.google'
+		let validateLink = function (value) {
+		  return /^(?:(?:(?:https?|ftp):)?\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,5})?(?:[/?#]\S*)?$/i.test(value);
+		}
+
+ 		// Parsing date function; NOT READY FOR DIFFERENT TIMEZONES
+		let parseDate = function (input) {
+		  let datetime = input.split('T');
+		  let date = datetime[0].split('-');
+		  let time = datetime[1].split(':');
+		  return new Date(date[0], date[1]-1, date[2], time[0], time[1], time[2]).toLocaleString();
+		}
+
+		// Get form, link input, submit input, message span and listing
+		let form = document.querySelector('form')
+		let input = document.getElementById('link')
+		let submit = document.getElementById('submit')
+		let help = document.getElementById('help')
+		let listing = document.querySelector('#listing tbody')
+
+		// Set name for currently used storage according to currently used version
+		let currentLocalStorage = "Link_Biscuit:v0"
+
+		// Initialize variables for records that will be listed
+		let records = new Array()
+		let numOfRecords = 0
+
+		// Also initialize 'recordsTmp' which serves as a temporary storage for control whether previously stored records were successfully merged with the new record
+		let recordsTmp = new Array()
+
+		// Reset currentLocalStorage when testing
+		localStorage.setItem(currentLocalStorage, "");
+
+		// Let's check whether localeStorage is already set and contains expected values
+		if (localStorage.getItem(currentLocalStorage) !== undefined && localStorage.getItem(currentLocalStorage) !== null && localStorage.getItem(currentLocalStorage) !== "") {

```

Comment:
maybe this could be simplified by using [double bang ](https://medium.com/@edplatomail/js-double-bang-or-the-not-operator-part-40e55d089bf0)operator?